apps/src/templates/sectionProgress/sectionProgressRedux.js

```diff
@@ -41,6 +42,21 @@ export const addScriptData = (scriptId, scriptData) => {
 export const addStudentLevelProgress = (scriptId, studentLevelProgress) => ({
   type: ADD_STUDENT_LEVEL_PROGRESS, scriptId, studentLevelProgress
 });
+export const addStudentLevelPairing = (scriptId, studentLevelPairing) => {
+  const data = studentLevelPairing;
+  const isValid = Object.keys(data).every((userId) => (
+   Object.keys(data[userId]).every((levelId) => (
+     typeof data[userId][levelId] === "boolean"
+   ))
+ ));
+ if (!isValid) {
+   throw new Error("Input is invalid");
+ }
+
+ return {
+  type: ADD_STUDENT_LEVEL_PAIRING, scriptId, studentLevelPairing
+  };
+};
```

Comment:
Looks like we need some whitespace adjustments here.  Did we want to put in a more helpful error message if our validation fails?